portal-kernel/src/com/liferay/portal/kernel/portlet/render/PortletRenderUtil.java

```diff
@@ -493,6 +509,11 @@ private static void _writeJavaScriptPath(
 		if (javaScriptPath.startsWith("module:")) {
 			javaScriptPath = javaScriptPath.substring(7);
 
+			type = "module";
+		}
+		else if (javaScriptPath.startsWith("module-shim:")) {
+			javaScriptPath = javaScriptPath.substring(12);
+
 			type = FrontendESMUtil.getScriptType();
```

Comment:
In other places we skipped the call to `FrontendESMUtil.getScriptType()` , just noting that here we don't

Reply:
Because what we are doing is to support a new type of qualifier in portlet-js headers in addition to `module`, which is `module-shim`. And in those cases where `module-shim` is used, we print whatever is configured in the portal in that moment: if shim is active, we print `module-shim`, otherwise we print `module` as usual.

Note that these `module` and `module-shim` syntax can be used in any portlet, not only in client extensions. Nobody has asked for that, but it's a feature emerging from this.